### DISCUSSION TITLE
Criar função sem setar variavel

### DISCUSSION THREAD

**Discussion by adolfomoro** [frontendbr/forum#1752] [Perguntas e Respostas]
Como eu crio uma função sem q depois ela se destroi, tipo
new Funcao('1');
Sem precisar colocar o var variavel = 

E depois puxar outra fucan dentro dela assim:

Funcao.Teste() e retornar o 1 q foi inserido

NODE

**Comment by vhfmag** [comment]
Algo assim?

```js
function gerarClosure() {
  let valorArmazenado;
  
  function armazenarValor(valor) {
    valorArmazenado = valor;
  }
  
  function pegarValor() {
    return valorArmazenado;
  }

  return { armazenarValor, pegarValor };
}

const { armazenarValor, pegarValor } = gerarClosure();

armazenarValor(1);
console.log(pegarValor()); // 1
```

Só por curiosidade, qual é seu caso de uso? :thinking:

**Comment by adolfomoro** [comment]
> Algo assim?
> 
> ```js
> function gerarClosure() {
>   let valorArmazenado;
>   
>   function armazenarValor(valor) {
>     valorArmazenado = valor;
>   }
>   
>   function pegarValor() {
>     return valorArmazenado;
>   }
> 
>   return { armazenarValor, pegarValor };
> }
> 
> const { armazenarValor, pegarValor } = gerarClosure();
> 
> armazenarValor(1);
> console.log(pegarValor()); // 1
> ```
> 
> Só por curiosidade, qual é seu caso de uso? 🤔

Entao, estou tentando criar um sistema de rotas, mas acho q vou desistir haha Esta bem dificil, pq as q existem n me atendem, estou tentando fazer outra coisa q tambem n estou conseguindo, acessar as funcao da pagina q foi carregada, vou tentar usar alguma existente msm

**Comment by ninetails** [comment]
Tenta ver mais sobre [Closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) e [callbacks](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function).

JavaScript trata funções como cidadões de primeira classe tais como objetos. Vc pode passar referências de funções como argumentos e retornos de funções ou estando referenciados em objetos e _prototypes_ (instâncias POO).
